src/hoc/auth-route.tsx

```diff
@@ -0,0 +1,40 @@
+import { GetServerSidePropsContext } from "next";
+import { store } from "../store";
+import { checkTokenValidity } from "../utils/token-validity-check";
+
+export interface AuthGetServerSidePropsContext
+  extends GetServerSidePropsContext {
+  data;
+}
+
+export const authGetServerSideProps = (getServerSideProps?: Function) => {
+  const Wrapper = async (ctx: AuthGetServerSidePropsContext) => {
```

Comment:
We can make it cleaner by just `return async (ctx: AuthGetServerSidePropsContext) => {`

Reply:
isn't returning async, not a good practice?

Reply:
In this case i think it is fine